### ISSUE TITLE
Feature request: Keep GPS on in order to improve accuracy before logging a point

### ISSUE THREAD

**Issue by jimis** [mendhak/gpslogger#654]
I've set the accuracy filter to 20m. But the phone's GPS can achieve accuracies up to 3m under ideal conditions AFAIK. As a result I get point after point with vastly different accuracies, resulting in a very misleading path if all points are connected.

Additionally, when I press "annotate" I expect an exact fix,and it's usually less accurate than the last automatic point.

I would like an option to "keep GPS on for X seconds after accuracy filter has been reached, in order to improve accuracy even further". For example
* after reaching the accuracy filter and storing in memory point A,
* it would keep logging in memory every 1s for the next X seconds and
* replace point A with the new point each time, only if the new point has a better accuracy;
* after X seconds, it would log the point to storage.

**Comment by iste** [comment]
👍 
Also, a feature "Accuracy goal" could also be great. 
This would also avoiding discarning every low accuracy points whiout have to deal with medium one.

**Comment by mendhak** [comment]
Sooooooo... I have no idea if anyone's still here but this accuracy topic is one of the most requested changes so I finally had some time to try doing something... 

![image](https://github.com/mendhak/gpslogger/assets/746276/08c1d7fe-2880-421b-8a67-12e831c2c413)

**tl;dr: if anyone is able to test, APK is here: https://github.com/mendhak/gpslogger/releases/tag/v130-rc1** 

There are multiple variations of the suggestions (#654 #591 #453) and many are quite complicated/detailed, I think I'm reaching a middle ground. 

The new option, when enabled: instead of matching on the _first_ point matching the user's accuracy, will instead keep trying for the accuracy duration, and pick the most accurate point. It matches jimis' description above. 

It isn't exactly the ability to log _any_ point, but can accomplish close enough if the accuracy filter is high, with a retry duration, which improves the chances of logging a good point. 

It isn't exactly the same as a _range_ either, eg "User wants 15m accuracy but will accept up to 40m" — I did try that but it made the code a lot more complex and the UI got confusing. Even this basic implementation is still quite a bit involved. The way to accomplish that would be to keep it at 40m but give it a decent retry time like 15seconds.

**Comment by mendhak** [comment]
v130 is now on FDroid and the releases. 

The new option uses the time period to pick the best location it can find. I didn't want to make it complicated by having a range, as there is already a lot going on in there.  This should help with jumpy lines and make things smoother at the expense of time. So you'll probably want to experiment here, eg, don't make the duration too long, maybe 5-10s will give it a better chance of getting a good point. 

![image](https://github.com/mendhak/gpslogger/assets/746276/25e434fc-c4af-4c0c-9c30-17447ac8ceaa)
